client/sites-dashboard/components/sites-grid.tsx

```diff
@@ -27,16 +27,31 @@ interface SitesGridProps {
 	className?: string;
 	isLoading: boolean;
 	sites: SiteExcerptData[];
+	siteSelectorMode?: boolean;
+	onSiteSelectBtnClick?: ( site: SiteExcerptData ) => void;
```

Comment:
I'd say `onSiteSelect` is a better name. Why highlight that this event was fired through a click of a button?

Reply:
Why not make the whole area clickable? It's very possible that the user might click the thumbnail instead of the button for example.

Reply:
I just followed the design:

<img width="438" alt="Screenshot 2023-05-16 at 20 41 37" src="https://github.com/Automattic/wp-calypso/assets/1241413/291c1c73-294a-4674-af1a-417b7c6d62f6">

If I rethink, I like having an explicit button for picking a site since the grid item component has options to provide more CTAs over the Thumbnail, such as plan renewal or similar.